src/main/java/com/textmessenger/security/UserPrincipal.java

```diff
@@ -0,0 +1,73 @@
+package com.textmessenger.security;
+
+import com.textmessenger.model.entity.User;
+import lombok.Data;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Data
+public class UserPrincipal implements UserDetails {
+
+  private Long id;
+
+  private String username;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused "username" private field. ([squid:S1068](https://sonarcloud.io/organizations/final_project_text_messenger/rules#rule_key=squid%3AS1068))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=Text_Messenger_Project&pullRequest=142&issues=AWWFkfhHX2sY87p0v-Gv&open=AWWFkfhHX2sY87p0v-Gv)

[ISSUE_KEY]: AWWFkfhHX2sY87p0v-Gv